moveit_planners/ompl/ompl_interface/src/parameterization/joint_space/constrained_planning_state_space.cpp

```diff
@@ -0,0 +1,108 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, KU Leuven
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Jeroen De Maeyer */
+
+#include <moveit/ompl_interface/parameterization/joint_space/constrained_planning_state_space.h>
+
+#include <ompl/base/spaces/constraint/ConstrainedStateSpace.h>
+
+const std::string ompl_interface::ConstrainedPlanningStateSpace::PARAMETERIZATION_TYPE =
+    "ConstrainedPlanningJointModel";
+
+ompl_interface::ConstrainedPlanningStateSpace::ConstrainedPlanningStateSpace(
+    const ModelBasedStateSpaceSpecification& spec)
+  : ModelBasedStateSpace(spec)
+{
+  setName(getName() + "_" + PARAMETERIZATION_TYPE);
+}
+
+double* ompl_interface::ConstrainedPlanningStateSpace::getValueAddressAtIndex(ompl::base::State* ompl_state,
+                                                                              const unsigned int index) const
+{
+  if (index >= variable_count_)
+    return nullptr;
+
+  return ompl_state->as<ompl_interface::ConstrainedPlanningStateSpace::StateType>()->values + index;
+
+  // (jeroendm) left in the debug code, because it is not unlikely that issues will arise here when adding new things in
+  // the future. Use dynamic casting to make our debugging live easier.
```

Comment:
You can also link to a writeup with debugging instructions instead

Reply:
Good idea, it is a bit ugly to leave this in.

Reply:
I replaced the long comment with: `// Developer tip: replace this with a dynamic_cast for debugging`. As for debugging instructions, I think it will be useful if I write a good description of all these details in a central place somewhere. But I need to come up with a good format and a good place to put it. Maybe the MoveIt [concepts page](https://moveit.ros.org/documentation/concepts/) on the website?